### DISCUSSION TITLE
Does it make sense to enforce positivity of the state variable in FL=103?

### DISCUSSION THREAD

**Discussion by Thomas-Ulrich** [SeisSol/SeisSol#746] [General]
I investigated why Taufiq's branch (Ridgecrest_2019) for running the Ridgecrest setup gives very different results than the latest master branch + double nucleation.
(the Southern portion of the main fault is not ruptured).

I've finally found that the differences are due to this change:
[https://github.com/SeisSol/SeisSol/compare/master...2019_Ridgecrest_sequence#diff-5d9afd7c003de5c508a8306392b41[…]5a840f9f7e0288d3ee8f849685f85L1076](https://github.com/SeisSol/SeisSol/compare/master...2019_Ridgecrest_sequence#diff-5d9afd7c003de5c508a8306392b41735eb75a840f9f7e0288d3ee8f849685f85L1076)
which revert changes of https://github.com/SeisSol/SeisSol/pull/306
If I remove the enforcement to have the state variable positive, rupture can break across the conjugate fault during the mainshock event in the latest master branch.

Here is the state variable (saturated colorbar) at t=25s of each simulation:
![image](https://user-images.githubusercontent.com/13536910/204743692-58859c0c-179e-4a0f-952f-05529913dea7.png)
(bottom is no positivity enforcement).

Removing the positivity enforcement may also fix https://github.com/SeisSol/SeisSol/issues/668
(and then we may need to revert the check for StV(t=0)>0 in FL=103.

Edit: the SCEC tpv104 benchmark does not lead to negative state variables.

**Comment by sebwolf-de** [comment]
Just to make sure, you removed this `max`, right? https://github.com/SeisSol/SeisSol/blob/master/src/DynamicRupture/FrictionLaws/FastVelocityWeakeningLaw.h#L134-L135
The same `max` appears also in the old Fortran version: https://github.com/SeisSol/SeisSol/blob/v0.9.0/src/Physics/Evaluate_friction_law.f90#L1087-L1089

**Reply by Thomas-Ulrich** [reply]
actually, what I did is to add back the max on Taufiq's branch (and then I get a final slip looking visually similar with what I get on the latest master branch).

**Comment by daisy20170101** [comment]
The state variable should be physically non-zero, right? I am wondering if a negative StV results in NaN in the log()?

**Reply by Thomas-Ulrich** [reply]
not it does not leads to nan in tpv103 (as you can see on Ridgecrest setup).
It only results in NaN for the slip and ageing law (according to the PR)

**Comment by daisy20170101** [comment]
But in one of my previous issue, it seems that adding the max() fix NaNs in the simulations with FL103 of me and James, https://github.com/SeisSol/SeisSol/pull/361 <https://github.com/SeisSol/SeisSol/pull/361>

**Reply by Thomas-Ulrich** [reply]
By doing that you fix the nan issue (which is probably due to floating point overflow, because of the exponential).
This does not prove that your fix is physically correct.
 Maybe there is an other way to fix the nan issue?

**Comment by daisy20170101** [comment]
Yes there might be others. 
According to your tests,  master branch generates the virtually same results as 2019_Ridgecrest branch FL103 + max() fix, right? 
After all, they converge to the same result?

**Reply by Thomas-Ulrich** [reply]
Here is a diff of RT, PSR and ASl:
![image](https://user-images.githubusercontent.com/13536910/204779637-648a68dd-e77c-4ca2-99f9-f37b7f5b295b.png)
Differences are overall small.
Might be associated with the bug fix on bimaterial conditions?

**Reply by NicoSchlw** [reply]
Which differences do you show? master without max() and 2019_ridgecrest_sequence?

**Reply by Thomas-Ulrich** [reply]
yes.
